### PR TITLE
Fix PositionComponent.scale bug

### PR DESCRIPTION
**Pull request by spydon**
# Description

The canvas should be translated with the anchor in regards to `size`, not the `scaledSize`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame/CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@
  - Introduce `scale` on `PositionComponent`
  - Add `ScaleEffect` that works on `scale` instead of `size`
  - Add class `NotifyingVector2`
+ - Fix `scale` in `PositionComponent`
```
Comment (wolfenrain) [inline packages/flame/CHANGELOG.md:32]: Just a general question, but this is a fix for a change that wasn't released yet, do we need to have a CHANGELOG entry then? As it doesn't "change" anything for the public facing user?

**Review by wolfenrain — APPROVED** [review]
LGTM, happy with more tests !

File: packages/flame/test/components/position_component_test.dart [hunk 2]
```diff
@@ -228,5 +228,61 @@ void main() {
       );
       expect(child.absoluteCenter, parent.position + child.center);
     });
+
+    test('scaled component contains point', () {
```
Comment (erickzanardo) [inline packages/flame/test/components/position_component_test.dart:232]: We should check our current coverage and increase it if we can 😉

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/CHANGELOG.md (hunk 1 above)
Comment (erickzanardo) [reply packages/flame/CHANGELOG.md:32]: I agree, no need for this entry

Comment (spydon) [reply packages/flame/CHANGELOG.md:32]: True, I'll remove it. Then it is not public facing that I broke it either. ;)

File: packages/flame/test/components/position_component_test.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/test/components/position_component_test.dart:232]: Good idea, increased it

**Pull request merged**